components/brave_wallet_ui/common/hooks/token.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as BraveWallet from 'gen/brave/components/brave_wallet/common/brave_wallet.mojom.m.js'
+import * as React from 'react'
+import {
+  FindERCTokenInfoReturnInfo,
+  ERCToken,
+  EthereumChain
+} from '../../constants/types'
+
+export default function useTokenInfo (
+  findERCTokenInfo: (address: string) => Promise<FindERCTokenInfoReturnInfo>,
+  visibleTokens: ERCToken[],
+  fullTokenList: ERCToken[],
+  selectedNetwork: EthereumChain
+) {
+  const [tokenContractAddress, setTokenContractAddress] = React.useState<string>('')
+  const [foundTokenInfoByContractAddress, setFoundTokenInfoByContractAddress] = React.useState<ERCToken | undefined>()
+
+  const onFindTokenInfoByContractAddress = (contractAddress: string) => {
+    setTokenContractAddress(contractAddress)
+  }
+
+  React.useEffect(() => {
```

Comment:
Instead of a `useEffect`, can we do a `useCallback` that takes `tokenContractAddress` as argument? `setTokenContractAddress` is not used anywhere else either.

Reply:
Left a comment here per our discussion/call of why we are using `useEffect` here rather than `useCallback`